pages/home/auth/secure-auth-production.mdx

```diff
@@ -0,0 +1,127 @@
+
+In a production application or agent, user sessions are verified by your code, not Arcade.dev. To enable this, build a custom verifier route and add the URL to the Arcade Dashboard.
+
+When your users authorize a tool, Arcade.dev will redirect the user's browser to your verifier route with some information in the query string. Your custom verifier route must send a response back to Arcade.dev to confirm the user's session details.
+
+If you need help, start a [GitHub discussion](https://github.com/ArcadeAI/arcade-ai/discussions) and we'll be happy to assist.
+
+import { Steps, Tabs } from "nextra/components";
+
+<Steps>
+
+### Build a new route
+
+Create a public route in your app or API that can accept a browser redirect (HTTP 303) from Arcade.dev after a user has authorized a tool.
+
+The route must gather the following information:
+
+- The `flow_id` from the current URL's query string
+- The unique ID of the user currently signed in. This varies depending on how your app is built, but it is typically retrieved from a session cookie or other secure storage. It **must** match the user ID that your code specified at the start of the authorization flow.
+
+
+### Verify the user's session details
+
+Use the Arcade SDK (or our REST API) to verify the user's session details.
+
+<Warning>
+  Because this request uses your Arcade API key, it *must not* be made from the client side (a browser or desktop app). This code must be run on a server.
+</Warning>
+
+<Tabs items={["REST"]} storageKey="preferredLanguage">
+<Tabs.Tab>
+```text
+POST https://cloud.arcade.dev/api/v1/oauth/confirm_user
+Authorization: Bearer <arcade_api_key>
+Content-Type: application/json
+
+{
+  "flow_id": "<flow_id from the query string>",
+  "user_id": "<the current user's ID>"
+}
+```
+</Tabs.Tab>
+</Tabs>
+
+**Valid Response**
+
+If the session details are valid, the response will contain some information about the auth flow:
+
+<Tabs items={["REST"]} storageKey="preferredLanguage">
+<Tabs.Tab>
+```text
+HTTP 200 OK
+Content-Type: application/json
+
+{
+  // Can be used to look up the auth flow's status in the Arcade API
+  "auth_id": "ac_2zKml...",
+
+  // Optional: URL to redirect the user to after the authorization flow is complete
+  "next_uri": "https://..."
+}
+```
+</Tabs.Tab>
+</Tabs>
+
+You can optionally redirect the user's browser to the `next_uri`, which will display a generic "success" page. Or, you can redirect to your application as needed.
+
+**Invalid Response**
+
+If the session details are invalid, it means the user is not the same user who started the authorization flow.
+
+<Tabs items={["REST"]} storageKey="preferredLanguage">
+<Tabs.Tab>
+```text
+HTTP 400 Bad Request
+Content-Type: application/json
+
+{
+  "code": 400,
+  "msg": "An error occurred during verification"
+}
+```
+</Tabs.Tab>
+</Tabs>
+
+### Add your custom verifier route to Arcade
+
+In the Arcade Dashboard, pick the **Custom verifier** option and add the URL of your verifier route.
+
```

Comment:
TODO: Add screenshot soon

Reply:
@torresmateo Will do, I will update this after the Dashboard ships.